dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -4302,15 +4325,33 @@ bool Sedp::TypeLookupReplyReader::process_get_types_reply(const XTypes::TypeLook
     ACE_DEBUG((LM_DEBUG, "(%P|%t) Sedp::TypeLookupReplyReader::process_get_types_reply\n"));
   }
 
+  if (reply._cxx_return.getType.return_code != DDS::RETCODE_OK) {
+    if (DCPS::DCPS_debug_level) {
+      ACE_DEBUG((LM_DEBUG, "(%P|%t) Sedp::TypeLookupReplyReader::process_get_types_reply - "
+                 "received reply with return code %C\n",
+                 DCPS::retcode_to_string(reply._cxx_return.getType.return_code)));
+    }
+    return false;
+  }
+
   if (reply._cxx_return.getType.result.types.length() == 0) {
     if (DCPS::DCPS_debug_level) {
       ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: Sedp::TypeLookupReplyReader::process_get_types_reply - "
-        "received reply with no data\n"));
+                 "received reply with no data\n"));
     }
     return false;
   }
 
   sedp_.type_lookup_service_->add_type_objects_to_cache(reply._cxx_return.getType.result.types);
+
+  if (reply._cxx_return.getType.result.complete_to_minimal.length() != 0) {
+    if (DCPS::DCPS_debug_level) {
```

Comment:
Should this be conditional on a higher debug level?

Reply:
Is debug level 2 (logs once per DDS entity) appropriate for this?

Reply:
Maybe 3 or 4?  It could be more than once per entity since it would be based on the associations with remote entities.  @iguessthislldo may have some thoughts on this